iceoryx_hoofs/testing/include/iceoryx_hoofs/testing/ctor_and_assignment_operator_test_class.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+// Copyright (c) 2023 by Mathias Kraus <elboberido@m-hias.de>. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_TESTING_CTOR_AND_ASSIGNMENT_OPERATOR_TEST_CLASS_HPP
+#define IOX_HOOFS_TESTING_CTOR_AND_ASSIGNMENT_OPERATOR_TEST_CLASS_HPP
+
+#include <cstdint>
+#include <vector>
+
+namespace iox
+{
+namespace testing
+{
+template <typename T = uint64_t, T DEFAULT_VALUE = 0>
+class CTorAndAssignmentOperatorTestClass
```

Comment:
@elBoberido I think one of those names would fit better:

1. `LifetimeVerificator`
2. `LifetimeTracker`
3. `CTorAssignmentCounterClass`
4. `StatisticalObjectMonitor`
5. `ObjectLifetimeInspector`

What do you think?

Reply:
I'm also not that happy with the name. I just labeled it according what it does :sweat_smile: 

What do you think of `LifetimeAndAssignmentTracker`?

I'll merge now but will happily change in a follow up

Reply:
Move discussion to the issue :)

Reply:
@elBoberido 

> What do you think of `LifetimeAndAssignmentTracker`?

Sounds good to me. Way better than what is currently used :stuck_out_tongue: